packages/graph/migrations/postgres/migration/1656417312397_initial.ts

```diff
@@ -364,10 +359,59 @@ export async function up(pgm: MigrationBuilder): Promise<void> {
       "link_type_version_id",
     ],
   });
-  pgm.addConstraint("links", "links_single_or_multi", {
-    // multi_order is only appropriate to set if multi is true.
-    check: "(multi) or (multi_order IS NULL)",
-  });
+
+  pgm.createTable(
+    "link_histories",
+    {
+      // We should consider whether these should reference entity_ids or not.
+      // If we allow GDPR removal of entities, this constraint has to fail/cascade depending on desired output.
+      source_entity_id: {
+        type: "UUID",
+        notNull: true,
+        references: "entity_ids",
+      },
+      target_entity_id: {
+        type: "UUID",
+        notNull: true,
+        references: "entity_ids",
+      },
+      link_type_version_id: {
+        type: "UUID",
+        notNull: true,
+        references: "link_types",
+      },
+      link_order: {
+        // TODO: this is where we could do fractional indexing
+        //  https://app.asana.com/0/1200211978612931/1202085856561975/f
+        type: "integer",
+        notNull: false,
+      },
+      created_by: {
+        type: "UUID",
+        notNull: true,
+        references: "accounts",
+      },
+      created_at: {
+        type: "TIMESTAMP WITH TIME ZONE",
+        notNull: true,
+      },
+      removed_by: {
+        type: "UUID",
+        notNull: true,
+        references: "accounts",
+      },
+      removed_at: {
+        type: "TIMESTAMP WITH TIME ZONE",
+        notNull: true,
+        // This default would give the UTC time "wall clock" even within a transaction/
+        default: pgm.func("clock_timestamp()"),
```

Comment:
Same as above

Reply:
Addressed in 708836711944a5a6b4f6c71215adbbb77b22b58d, had to add explicit `clock_timestamp` to `created_by` in 7e6e1d2908e5fad4556c028d63aab737a7cc7982